### PR TITLE
OADP-205: Remove registry deployment in favor of plugin enhancement

### PR DESCRIPTION
**Pull request by kaovilai**
This velero plugin registry enhancements will do the following
- [x] Find registry deployments for dpa BSLs and delete them
- [x] Set additional environment variable to velero deployment to enable plugin registry behavior
  - [x]  This can only be tested in CI once https://github.com/openshift/openshift-velero-plugin/pull/145 is merged and quay.io image is updated.
  
Install this PR operator with
```
operator-sdk run bundle quay.io/tkaovila/oadp-operator-bundle:velero-plugin-registry --namespace openshift-adp
```

Instruction for testing this PR prior to updated latest image at https://github.com/openshift/openshift-velero-plugin/pull/145

[OADP-205](https://issues.redhat.com//browse/OADP-205)

### CODE REVIEW

**Comment by kaovilai** [comment]
oops forgot needs plugin PR merge first

**Comment by kaovilai** [comment]
ready for review but
/hold for https://github.com/openshift/openshift-velero-plugin/pull/145

**Comment by kaovilai** [comment]
ready for review but
/hold for https://github.com/openshift/openshift-velero-plugin/pull/145

File: controllers/velero.go
Comment (shubham-pampattiwar) [inline controllers/velero.go:584]: +1 on the usage on ENV Var to trigger plugin registry.

File: tests/e2e/backup_restore_suite_test.go
Comment (shubham-pampattiwar) [inline tests/e2e/backup_restore_suite_test.go:126]: Remove the commented code ?

File: controllers/registry.go
Comment (shubham-pampattiwar) [inline controllers/registry.go:211]: Suggestion: Add the name and namespace of the registry deployment deleted in the Event instead of a generic statement.

**Review by shubham-pampattiwar — COMMENTED** [review]
Operator changes look straight forward to me, added some NITs, Thanks @kaovilai

**Review by dymurray — CHANGES_REQUESTED** [review]
This PR also needs to include changes to the CSV to remove the registry as a related image.

**Comment by kaovilai** [comment]
bundle is up to date

File: controllers/registry.go
Comment (kaovilai) [reply controllers/registry.go:206]: IIUC the object will be attached to the event anyways. I'll add name and namespace to the string.
```go
        // Event constructs an event from the given information and puts it in the queue for sending.
	// 'object' is the object this event is about. Event will make a reference-- or you may also
	// pass a reference to the object directly.
	// 'type' of this event, and can be one of Normal, Warning. New types could be added in future
	// 'reason' is the reason this event is generated. 'reason' should be short and unique; it
	// should be in UpperCamelCase format (starting with a capital letter). "reason" will be used
	// to automate handling of events, so imagine people writing switch statements to handle them.
	// You want to make that easy.
	// 'message' is intended to be human readable.
	//
	// The resulting event will be created in the same namespace as the reference object.
	Event(object runtime.Object, eventtype, reason, message string)
```

**Comment by kaovilai** [comment]
For AWS:
Internal 500 error
```
trying to reuse blob sha256:ca753d5e70afb765e08ae8e79316b2feb7618fc6c779109ac7f797709c51d9b6 at destination: failed to read from destination repository mongo-persistent/todolist-mongo-go: 500 (Internal Server Error)
```
For GCP
It appears GCS storage driver init function is not called... 
```
plugin panicked: StorageDriver not registered: gcs
```
https://github.com/distribution/distribution/blob/3f4c558dac40dfb4fe13285ffc43a0d738d3a3f0/registry/storage/driver/gcs/gcs.go#L80
For Azure envs used was malformed.
```
azure: malformed storage account key: illegal base64 data at input byte 0
```

**Comment by kaovilai** [comment]
https://github.com/openshift/openshift-velero-plugin/issues/152 might be an issue.

**Comment by kaovilai** [comment]
waiting on https://app.travis-ci.com/github/openshift/openshift-velero-plugin/builds/253057305 to retest again

**Comment by kaovilai** [comment]
Looks like no retest required.. since all running tests are still installing ocp

**Review by dymurray — COMMENTED** [review]
Code changes LGTM, will ack when E2E is updated to disable iamge backup for GCP

**Comment by kaovilai** [comment]
GCP tests blocked by https://github.com/openshift/openshift-velero-plugin/pull/155. Once travis builds this, retesting GCP is expected to pass.

**Review by dymurray — APPROVED** [review]

**Review by savitharaghunathan — COMMENTED** [review]
/lgtm

**Pull request merged**
